src/highdicom/pr/content.py

```diff
@@ -0,0 +1,876 @@
+                )
+            ref_im_item = Dataset()
+            ref_im_item.ReferencedSOPClassUID = ref_im.SOPClassUID
+            ref_im_item.ReferencedSOPInstanceUID = ref_im.SOPInstanceUID
+
+            if referenced_frame_number is not None:
+
+                def check_frame_number(f: int) -> None:
+                    if f < 1:
+                        raise ValueError(
+                            'Frame numbers must be positive integers'
+                        )
+                    elif f > ref_im.NumberOfFrames:
+                        raise ValueError(
+                            f'Frame number {f} is invalid for image with '
+                            f'{ref_im.NumberOfFrames} frames.'
+                        )
+
+                if isinstance(referenced_frame_number, Sequence):
+                    for f in referenced_frame_number:
+                        check_frame_number(f)
+                else:
+                    check_frame_number(referenced_frame_number)
+
+                ref_im_item.ReferencedFrameNumber = referenced_frame_number
+
+            if referenced_segment_number is not None:
+                n_segments = len(ref_im.SegmentSequence)
+
+                def check_segment_number(f: int) -> None:
+                    if f < 1:
+                        raise ValueError(
+                            'Segment numbers must be positive integers'
+                        )
+                    elif f > n_segments:
+                        raise ValueError(
+                            f'Segment number {f} is invalid for image with '
+                            f'{n_segments} segments.'
+                        )
+
+                if isinstance(referenced_segment_number, Sequence):
+                    for f in referenced_segment_number:
+                        check_segment_number(f)
+                else:
+                    check_segment_number(referenced_segment_number)
+
+                ref_im_item.ReferencedSegmentNumber = referenced_segment_number
+
+            ref_im_seq.append(ref_im_item)
+        self.ReferencedImageSequence = ref_im_seq
+
+        have_graphics = graphic_objects is not None and len(graphic_objects) > 0
+        have_text = text_objects is not None and len(text_objects) > 0
+        if not have_graphics and not have_text:
+            raise TypeError(
+                'Either graphic_objects or text_objects must contain at least '
+                'one item.'
+            )
+        if have_graphics:
+            for go in graphic_objects:
+                if not isinstance(go, GraphicObject):
+                    raise TypeError(
+                        'All items in graphic_objects must be of type '
+                        'highdicom.pr.GraphicObject'
+                    )
+                if go.units == AnnotationUnitsValues.MATRIX:
+                    sm_uid = VLWholeSlideMicroscopyImageStorage
+                    if referenced_images[0].SOPClassUID != sm_uid:
```

Comment:
This is explicitly stated in the standard as the only case where this is valid, thought I take your point that there are probably others. That said, I imagine the use case for presentation states of segmentations or parametric maps is likely vanishingly small at the moment.

Given that at this point in the code the referenced image datasets are available, would you like to suggest a way to programmatically determine whether an image is tiled (since not all segmentations and parametric maps will be it seems to make more sense to check the instance rather than the class)

Reply:
This is really a shortcoming of the standard and it should be corrected there. We will definitely need presentation states of parametric maps.

I define a tiled images as an image that contains the following attributes:

1. Number of Frames (even if there is only one frame, guaranteed to be included in Segmentation and Parametric Map instance)
2. Total Pixel Matrix Columns
3. Total Pixel Matrix Rows

(2) and (3) are already included by *highdicom* upon construction of Segmentation and Parametric Map instances.

Reply:
See also https://github.com/herrmannlab/highdicom/pull/139#discussion_r810514587

Reply:
ok, implemented a new utility check and applied it here. see 11aec6c7f6b306617c68441bd09963ec21f36600 and 3b3b8a117cdf733daa081c2ce408c3e912649f8a

Reply:
and 836e787 sorry I really messed up these commits